tests/unit/test_utils.py

```diff
@@ -665,3 +665,28 @@ def test_split_auth_from_netloc(netloc, expected):
 def test_remove_auth_from_url(auth_url, expected_url):
     url = remove_auth_from_url(auth_url)
     assert url == expected_url
+
+
+HAS_LEADING_DIR_TESTS = (
+    ('', False),
+    ('setup.py', False),
+    ('dir/', False),
+    ('pkg-1.0/setup.py', True),
+    ('''
+     setup.py
+     module.py
+     ''', False),
+    ('''
+     pkg-1.0/setup.py
+     pkg-1.0/module.py
+     ''', True),
+    ('''
+     a/setup.py
+     b/module.py
+     ''', False),
```

Comment:
What about the edge cases “pkg-1.0 pkg-1.0/“,  “pkg-1.0/ pkg-1.0/“, and “pkg-1.0/ pkg-1.0/setup.py”?